src/core/aws-net-websocket-registry.adb

```diff
@@ -1401,7 +1401,7 @@ package body AWS.Net.WebSocket.Registry is
       Timeout      : Duration := Forever;
       Asynchronous : Boolean := False)
    is
-      A : String (Natural (Message'First) .. Natural (Message'Last))
+      A : String (1 .. Natural (Message'Length))
```

Comment:
No need conversion to Natural here. 'Length compatible with all types as I can remember.

Reply:
In fact, this piece of code is potentially dangerous. What if you receive a message more than 2MB ? you'll blow up the stack and end up with Stack_Overflow.

Reply:
> No need conversion to Natural here. 'Length compatible with all types as I can remember.

Indeed. Now fixed.

Reply:
> In fact, this piece of code is potentially dangerous. What if you receive a message more than 2MB ? you'll blow up the stack and end up with Stack_Overflow.

I don't think it is the case here as the variable A is imported and overlaid on top of current parameter. So at the end it does not take any additional byte on the stack.

Reply:
I do not understand the "is imported" notion. You are declaring a new array on the stack. It has nothing to do with the parameters that the subprogram receives (except for looking at the bounds). So it will blow up. I actually fixed a similar case in my local copy of AWS (I don't remember whether it was this specific piece of code or another place, though). Does the AWS testsuite have a test where it sends a 5MB message, and another where it receives a 5MB message ?

Reply:
It is not only imported but it has an Address clause.

Reply:
@briot : You can test it by yourself, even 150 Mb is passing without an exception. There is nothing put on the stack because of the Address clauses.

```
with Ada.Text_IO;
with Ada.Streams;

procedure Tmem is

   use Ada.Text_IO;
   use Ada.Streams;

   procedure Send (P : String) is
   begin
      for C of P loop
         if C /= 'x' then
            Put_Line ("error!");
         end if;
      end loop;
   end Send;

   procedure Send (P : Stream_Element_Array) is
      A : String (1 .. P'Length) with Address => P'Address;
   begin
      Send (A);
   end Send;

   type SA is access Stream_Element_Array;

   P : SA := new Stream_Element_Array (1 .. 150 * 1_024 * 1_024);

begin
   for I in P'range loop
      P (I) := Character'Pos ('x');
   end loop;

   Send (P.all);

   Put_Line ("All good !");
end Tmem;
```